vscode-lean4/src/infoview.ts

```diff
@@ -248,6 +254,7 @@ export class InfoProvider implements Disposable {
 
     private async sendPosition() {
         if (!window.activeTextEditor || !languages.match(this.leanDocs, window.activeTextEditor.document)) { return null; }
+        void this.autoOpen();
```

Comment:
Why add this here?

Reply:
if user changes this preference then next time they move the position in the editor the infoview will popup.

![image](https://user-images.githubusercontent.com/18707114/135522084-2240d5d7-650b-4b6f-bbff-3a04f04fe0da.png)

Reply:
Ah, then maybe it should go into the `workspace.onDidChangeConfiguration` hook? If that ends up starting the infoview too soon (i.e. while the user is still changing settings) then let's keep it in `sendPosition` but maybe rename it to `onPositionChanged` or some such since it now does other things besides just sending the position.

Reply:
Yeah, exactly, that felt too soon, and it crowds out the settings editor...

Reply:
> it crowds out the settings editor...

If you're in the settings editor, then the infoview wouldn't open (afaict, since it's the active editor and not a lean file).

It took me a while to understand what the actual issue here was.  The PR fixes two things:
 - The infoview reopens when the Lean server is restarted,
 - and it opens for markdown files.

Please just note this in the PR description / commit message.  We should also update the description of the setting to reflect the current behavior.

I think it would be clearer to add the `autoOpen` handler to the `onDidChangeActiveTextEditor` handler.

Reply:
Ok, I tried moving it to onDidChangeActiveTextEditor , but it is happening too soon (before LeanClient calls languages.setTextDocumentLanguage(doc, 'lean4')), so I added a new didSetLanguage event to the client which the InfoView listens to.  Now the following scenario works:
1. open vs code on lean package with setting AutoOpen turned off
2. open a .lean source file (note: no InfoView), but lean server is started.
3. close the .lean source file
4. Change the AutoOpen setting to Checked.
5. Open a .lean source file (InfoView pops up when it realizes this is a lean4 document).